pkg/cnb/remote_stack_reader.go

```diff
@@ -25,13 +25,17 @@ type RemoteStackReader struct {
 	Keychain       authn.Keychain
 }
 
-func (r *RemoteStackReader) Read(clusterStackSpec v1alpha1.ClusterStackSpec) (v1alpha1.ResolvedClusterStack, error) {
-	buildImage, buildIdentifier, err := r.RegistryClient.Fetch(r.Keychain, clusterStackSpec.BuildImage.Image)
+func (r *RemoteStackReader) Read(keychain *authn.Keychain, clusterStackSpec v1alpha1.ClusterStackSpec) (v1alpha1.ResolvedClusterStack, error) {
+	if keychain == nil {
+		keychain = &r.Keychain
+	}
+
+	buildImage, buildIdentifier, err := r.RegistryClient.Fetch(*keychain, clusterStackSpec.BuildImage.Image)
 	if err != nil {
 		return v1alpha1.ResolvedClusterStack{}, err
 	}
 
-	runImage, runIdentifier, err := r.RegistryClient.Fetch(r.Keychain, clusterStackSpec.RunImage.Image)
+	runImage, runIdentifier, err := r.RegistryClient.Fetch(*keychain, clusterStackSpec.RunImage.Image)
```

Comment:
We did it like this to make consistent with the `Builder`. It uses the `KeyChainFactory` in the `Reconciler` to read the keychain and hand it over to the `BuilderCreater`.

We tried to do the same for `ClusterStore` and `ClusterStack`. Use the `KeyChainFactory` in the `Reconciler` and hand it over to `ClusterStackReader` or `ClusterStoreReader`.

I think that this would be more consistent and therefor preferable. WDYT?

Reply:
That makes sense to me.